react/week4/github-user-searcher-app/src/App.js

```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+import './App.css';
+import Root from './Components/Root';
+import Context from './DefaultContext';
+
+function App() {
+  const [searchToken, setSearchToken] = useState('');
+  const [status, setStatus] = useState('');
+  const [searchType, setSearchType] = useState('namesearch');
+  return (
```

Comment:
Your provider logic should be extracted to a different component. For that you can use the `children`. The result should be something like

```jsx
function App() {
 return (
   <GitHubSearchProvider> 
      <Root />
   </GtHubSearchProvider>
)
}
```

You can accomplish that by passing a children prop to the `GitHubSearchProvider` as you have learned in previous classes.

Reply:
Very good point